src/appservice/Appservice.ts

```diff
@@ -225,6 +226,19 @@ export class Appservice extends EventEmitter {
             throw new Error("Expected user namespace to be a prefix");
         }
         this.userPrefix = this.userPrefix.substring(0, this.userPrefix.length - 2); // trim off the .* part
+
+        if (!this.registration.namespaces || !this.registration.namespaces.aliases || this.registration.namespaces.aliases.length === 0) {
+            throw new Error("No alias namespaces in registration");
```

Comment:
this is a valid condition, and part of the reason for why the alias prefixing stuff was never added. This error should be raised at each point the prefix is used rather than here.

Reply:
out of curiosity: how come the alias behaviour is supposed to be different from the user id one?

Reply:
The SDK isn't just for bridges, it's for all appservices. The user ID behaviour is required in order for any of the SDK operations to work correctly whereas the aliases are not critical to most appservices or the SDK operations.